game/src/info/trip.rs

```diff
@@ -386,6 +389,329 @@ pub fn cancelled(ctx: &mut EventCtx, app: &App, id: TripID) -> Widget {
+            } else if time < &slow_delay_time {
+                (Color::BLACK, app.cs.slow_intersection)
+            } else {
+                (Color::WHITE, app.cs.very_slow_intersection)
+            };
+            // Draw label after, to prevent being obscured by the line
+            let time_duration = Duration::seconds(*time as f64);
+            let unzoomed_label = Text::from(TextSpan::from(
+                Line(format!("{}", time_duration)).body().fg(fg_color),
+            ))
+            .bg(bg_color);
+            let zoomed_label = Text::from(TextSpan::from(
+                Line(format!("{}", time_duration)).size(6).fg(fg_color),
+            ))
+            .bg(bg_color);
+            let unzoomed_rendered = unzoomed_label
+                .render(ctx)
+                .centered_on(intersection.polygon.center());
+            let zoomed_rendered = zoomed_label
+                .render(ctx)
+                .centered_on(intersection.polygon.center());
+            details.unzoomed.append(unzoomed_rendered.clone());
+            details.zoomed.append(zoomed_rendered);
+        }
+    }
+}
+
+/// Highlights lanes which were "slow" on the map
+fn highlight_slow_lanes(app: &App, details: &mut Details, id: TripID, ctx: &EventCtx) {
+    let data_opt = &app.primary.sim.get_analytics().trip_lane_speeds(id);
+    if data_opt.is_some() {
+        let data = data_opt.unwrap();
+        for (id, speed_percent) in data {
+            let lane = app.primary.map.get_l(*id);
+            let (fg_color, bg_color) = if speed_percent > &95 {
+                (Color::WHITE, app.cs.normal_slow_intersection)
+            } else if speed_percent > &60 {
+                (Color::BLACK, app.cs.slow_intersection)
+            } else {
+                (Color::WHITE, app.cs.very_slow_intersection)
+            };
+            details.unzoomed.push(
+                Fill::Color(bg_color),
+                lane.lane_center_pts.make_polygons(Distance::meters(10.0)),
+            );
+            details.zoomed.extend(
+                Fill::Color(bg_color),
+                lane.lane_center_pts.dashed_lines(
+                    Distance::meters(0.75),
+                    Distance::meters(1.0),
+                    Distance::meters(0.4),
+                ),
+            );
+            // Draw label after, to prevent being obscured by the line
+            let unzoomed_label = Text::from(TextSpan::from(
+                Line(format!("{}s", speed_percent)).body().fg(fg_color),
+            ))
+            .bg(bg_color);
+            let zoomed_label = Text::from(TextSpan::from(
+                Line(format!("{}s", speed_percent)).small().fg(fg_color),
+            ))
+            .bg(bg_color);
+
+            let (pt, _) = lane
+                .lane_center_pts
+                .must_dist_along(lane.lane_center_pts.length() / 2.0);
+            let unzoomed_rendered = unzoomed_label.render(ctx).centered_on(pt); //.rotate(angle);
```

Comment:
Style: Remove the commented rotation. I agree that these labels shouldn't be rotated to match the road; I think they're much easier to read as they are now, always oriented normally.

Reply:
Yup, again something I forgot to remove. Thanks for the spot!